app/static/src/app/store/graphs/actions.ts

```diff
@@ -0,0 +1,25 @@
+import { ActionTree } from "vuex";
+import { GraphsMutation } from "./mutations";
+import { AppState, AppType } from "../appState/state";
+import { FitDataAction } from "../fitData/actions";
+import { GraphsState } from "./state";
+
+export enum GraphsAction {
+    UpdateSelectedVariables = "UpdateSelectedVariables"
+}
+
+export const actions: ActionTree<GraphsState, AppState> = {
+    UpdateSelectedVariables(context, payload: { index: number; selectedVariables: string[] }) {
+        const { commit, dispatch, rootState } = context;
+        // Maintain unselected variables too, so we know which variables had been explicitly unselected when model
```

Comment:
sorry just wrapping my head around this... whats the reason for unselected variables? Can we just remove from selected list when they unselect? Seems like double info from my naive eye

Reply:
We keep track of unselected variables because when the model updates, we want to select any new variables (for the first graph) by default, but we also want to continue excluding any variables which were explicitly unselected. And I think by the time we got to making those selections post-model update, it wasn't convenient to have access to the old model's variable set, so explicitly holding onto unselectedVariables was the easiest way to do this. 

Arguably, we only need to hold on to unselected variables for the first graph, since we only add variables by default to the first one. But we're going to make it so that any graph can become the first one by deleting others, so I think it's probably best to just hang onto unselected for all.